packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -1849,12 +1868,8 @@ abstract class MultiSelectableSelectionContainerDelegate extends SelectionContai
     if (selections.isEmpty) {
       return null;
     }
-    final StringBuffer buffer = StringBuffer();
-    for (final SelectedContent selection in selections) {
-      buffer.write(selection.plainText);
-    }
     return SelectedContent(
-      plainText: buffer.toString(),
+      plainText: selections.map((SelectedContent e) => e.plainText).join(selectedContentSeparator),
```

Comment:
Just as a thought, what if instead of passing in a separator String, it was a function:

```
String Function(List<SelectedContent> selections)
```

Maybe this is overkill, but I wanted to talk through it.

Reply:
I was also thinking that having a way for the separator to not be the same for the entire container might be a good idea